openassessment/xblock/submission_mixin.py

```diff
@@ -203,19 +206,22 @@ def save_files_descriptions(self, data, suffix=''):  # pylint: disable=unused-ar
 
         Args:
             data (dict): Data should have a single key 'descriptions' that contains
-                the texts for each uploaded file.
+                a list of dictionaries with the following keys: 'description' and fileName.
+            each element of the list maps to a single file
             suffix (str): Not used.
 
         Returns:
             dict: Contains a bool 'success' and unicode string 'msg'.
         """
         if 'descriptions' in data:
-            descriptions = data['descriptions']
+            file_data = data['descriptions']
 
-            if isinstance(descriptions, list) and all([isinstance(description, six.string_types) for description in descriptions]):
+            if isinstance(file_data, list) and all([isinstance(description['description'], six.string_types) and
```

Comment:
Nit: you can break this across lines in a more readable way
```python
if isinstance(file_data, list) and all([
    isinstance(description['description'], six.string_types) and
    isinstance(description['fileName'], six.string_types)
    for description in file_data
]):
```

Reply:
done